### PR TITLE
refactor!: Seeding config simplification

### PR DESCRIPTION
**Pull request by paulgessinger**
This PR removes the dedicated config struct in `SeedingAlgorithm` in favor of it containing three config structs for the grid, seed finder and seed filter. Users of the seeding algorithm need to make sure that these config structs are configured consistently. `SeedingAlgorithm`'s constructor runs a number of checks to see if the config seems to make sense.

I tested the seeding example with this, and it gives the same number of seeds for a few particle gun events as main.

This is rebased on top of #941, so should be reviewed/merged after that.

### CODE REVIEW

File: Examples/Run/Reconstruction/Common/SeedingExample.cpp [hunk 1]
```diff
@@ -121,21 +121,43 @@ int runSeedingExample(int argc, char* argv[],
...
+  seedingCfg.seedFinderConfig.zMax = seedingCfg.gridConfig.zMax;
+
+  seedingCfg.seedFilterConfig.maxSeedsPerSpM = 1;
+  seedingCfg.seedFinderConfig.maxSeedsPerSpM =
+      seedingCfg.seedFilterConfig.maxSeedsPerSpM;
+
+  seedingCfg.gridConfig.cotThetaMax = 7.40627;  // 2.7 eta
+  seedingCfg.seedFinderConfig.cotThetaMax = seedingCfg.gridConfig.cotThetaMax;
+
+  seedingCfg.seedFinderConfig.sigmaScattering = 50;
```
Comment (robertlangenberg) [inline Examples/Run/Reconstruction/Common/SeedingExample.cpp:148]:
```suggestion
  seedingCfg.seedFinderConfig.sigmaScattering = 5;
```

File: Examples/Run/Reconstruction/Common/RecCKFTracks.cpp [hunk 2]
```diff
@@ -172,21 +172,46 @@ int runRecCKFTracks(int argc, char* argv[],
...
+
+      seedingCfg.seedFilterConfig.maxSeedsPerSpM = 1;
+      seedingCfg.seedFinderConfig.maxSeedsPerSpM =
+          seedingCfg.seedFilterConfig.maxSeedsPerSpM;
+
+      seedingCfg.gridConfig.cotThetaMax = 7.40627;  // 2.7 eta
+      seedingCfg.seedFinderConfig.cotThetaMax =
+          seedingCfg.gridConfig.cotThetaMax;
+
+      seedingCfg.seedFinderConfig.sigmaScattering = 50;
```
Comment (robertlangenberg) [inline Examples/Run/Reconstruction/Common/RecCKFTracks.cpp:202]:
```suggestion
      seedingCfg.seedFinderConfig.sigmaScattering = 5;
```

**Review by robertlangenberg — APPROVED** [review]
If I didn't miss anything due to similarities with !941 this also looks good. Since the grid doesn't actually take into account the scattering and impact parameters, this disallows hacks to make the grid coarser, i.e. this'll have to be fixed soon as well.

File: Examples/Run/Reconstruction/Common/SeedingExample.cpp (hunk 1 above)
Comment (paulgessinger) [reply Examples/Run/Reconstruction/Common/SeedingExample.cpp:148]: I think I would rather decouple changing this value from this PR.

File: Examples/Run/Reconstruction/Common/RecCKFTracks.cpp (hunk 2 above)
Comment (paulgessinger) [reply Examples/Run/Reconstruction/Common/RecCKFTracks.cpp:202]: Same as above.

File: Examples/Run/Reconstruction/Common/SeedingExample.cpp (hunk 1 above)
Comment (asalzburger) [reply Examples/Run/Reconstruction/Common/SeedingExample.cpp:148]: I agree - besides, we will probably find out that one single value for this is not good enough.

**Comment by asalzburger** [comment]
I will update the branch & then follow up later.

**Comment by asalzburger** [comment]
This was approved by @robertlangenberg  - merging.

**Comment by asalzburger** [comment]
@paulgessinger  - can you resolve the conflict now please after #941 went in?

**Comment by paulgessinger** [comment]
Done

**Pull request merged**